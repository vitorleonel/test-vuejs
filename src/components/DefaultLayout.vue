<script lang="ts">
import StyledHeader from '@/components/StyledHeader.vue'

export default {
  components: {
    StyledHeader
  }
}
</script>

<template>
  <StyledHeader />

  <main>
    <slot></slot>
  </main>
</template>

<style lang="scss" scoped>
main {
  padding-top: 32px;
  padding-bottom: 32px;
}
</style>
