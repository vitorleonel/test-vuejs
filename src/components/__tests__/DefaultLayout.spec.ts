import { mount, type VueWrapper } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

import DefaultLayout from '@/components/DefaultLayout.vue'

describe('DefaultLayout', () => {
  it('should render properly', () => {
    const wrapper: VueWrapper = mount(DefaultLayout, {
      slots: {
        default: '<p>Test content</p>'
      },
      global: { stubs: ['RouterLink'] }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
