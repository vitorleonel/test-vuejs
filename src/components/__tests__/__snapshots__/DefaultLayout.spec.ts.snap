// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DefaultLayout > should render properly 1`] = `
<div
  data-v-app=""
>
  
  <header
    class="header"
    data-v-51353816=""
    data-v-81cf18a8=""
  >
    <div
      class="container"
      data-v-51353816=""
    >
      <h1
        data-v-51353816=""
      >
        BestBuy
      </h1>
      <nav
        data-v-51353816=""
      >
        <router-link
          data-v-51353816=""
          to="/"
        >
          Home
        </router-link>
        <router-link
          data-v-51353816=""
          to="/shop"
        >
          Shop
        </router-link>
        <router-link
          data-v-51353816=""
          to="/about"
        >
          About us
        </router-link>
      </nav>
    </div>
  </header>
  <main
    data-v-81cf18a8=""
  >
    
    <p>
      Test content
    </p>
    
  </main>
  
</div>
`;
